system-tests/lib/cre/environment/environment.go

```diff
@@ -760,26 +624,15 @@ func SetupTestEnvironment(
 		fmt.Print(libformat.PurpleText("%s", stageGen.WrapAndNext("Wrote bootstrapping data into disk in %.2f seconds", stageGen.Elapsed().Seconds())))
 	}
 
-	// block on background stages
-	backgroundStagesWaitGroup.Wait()
-	close(backgroundStagesCh)
-
-	for result := range backgroundStagesCh {
-		if result.err != nil {
-			return nil, pkgerrors.Wrap(result.err, "background stage failed")
-		}
-		if result.panicValue != nil {
-			// Print the original stack trace from the background goroutine
-			if result.panicStack != nil {
-				fmt.Fprintf(os.Stderr, "Original panic stack trace from background goroutine:\n%s\n", result.panicStack)
-			}
-			panic(result.panicValue)
-		}
-		fmt.Print(result.successMessage)
+	if err := backgroundWorker.Wait(); err != nil {
+		return nil, pkgerrors.Wrap(err, "background worker failed")
 	}
 
+	// we ignore the error, because if that task would have failed, backgroundWorker.Wait() would have returned that error already
+	workflowRegistryConfigurationOutput, _ := workflowRegistrationResult.Get()
```

Comment:
IIUC result is a future, so I'd expect Get() to wait for me -- have I understood correctly? If so, why do I have to call backgroundWorker.Wait() explicitly?

Reply:
`Wait()` waiter for all tasks to complete